src/dali/abstract_ccxt_input_plugin.py

```diff
@@ -145,8 +145,12 @@ def _get_markets(self) -> List[str]:
 
     @staticmethod
     def _rp2_timestamp_from_ms_epoch(epoch_timestamp: str) -> str:
-        rp2_time = datetime.fromtimestamp((int(epoch_timestamp) / _MS_IN_SECOND), timezone.utc)
+        epoch_in_seconds, dummy_epoch_ms = divmod(float(epoch_timestamp), _MS_IN_SECOND)
```

Comment:
I think I mentioned before that this can be achieved with integer division:
```
epoch_in_seconds: int = int(epoch_timestamp) // _MS_IN_SECOND
```

Reply:
I understand, but I responded to that original comment My position was it was ok. I didn't receive any feedback.

Reply:
Sorry, at this point, there are so many comments and responses to this PR it is hard for me to keep track of it all.

The change just makes for cleaner code that's all, but if it's too much I guess I can do it when I do aliases.

Reply:
I understand. There was also issue my comments kept in pending state until I learned about submitting them :(. 

It's not too much and if there are strong feelings I don't mind simplifying it. In my original comment, my opinion is to not lose information such as truncating the ms part of the time. So I kept it as a placeholder if we have a change of heart. I think this is coming from when I recall three trades I had that all looked the same but were all unique but I couldn't tell because they were all in the same second. I later recognized that it was a bot that made 3 different trades in the sub-second resolution. So I just err on the side of if we have it not to lose it.

Reply:
The information isn't really needed though. Currently, RP2 only needs accuracy seconds. All transactions are recorded/reported that way, so the accuracy will eventually get lost anyway. In terms of taxes, that's all that is really necessary, considering pricing is only really available for minutes if it isn't provided by the exchange directly.

As for the trades happening within microseconds of each other, the best way to keep track of them is with unique ids.

@eprbell, what do you think about this fringe case? 
There is a fringe case where a user could make transactions (via bots) that if processed in the incorrect order could result in a negative balance if microsecond data is not preserved. For example,

Starting Balance 100.00 USDT
at 0100 - buy BTC with 100.00 USDT
at 0300 - sell BTC for 100.00 USDT
at 0500 - buy BTC with 100.00 USDT

If the two buys were processed first it would result in an error since the lot wouldn't exist. The transaction resolver processes each transaction individually, right? Could we group all transactions together that occurred in the same second for lot resolution? These transactions would still be reported individually of course.

I seriously doubt this would ever happen in the real world, but, in theory, it could.

Reply:
RP2 doesn't have precision limits on timestamps: it will parse whatever it is fed with microsecond, or even sub-ms precision (see crypto_example.ods / BTC / line 26 for an example). So it's really up to the exchanges (or the input preparer) to make sure the transactions are in an order that makes sense and have adequate time resolution. I would wait until we bump into the problem before we put together a solution for it.

Reply:
I appreciate the feedback good sirs.